core/src/test/java/com/klaytn/caver/scenario/RoleBasedAccountScenario.java

```diff
@@ -0,0 +1,396 @@
+        private String address;
+
+        public RoleBasedAccountGenerator() throws Exception {
+            transactionAccountCredential = new ArrayList<>();
+            updateAccountCredential = new ArrayList<>();
+            feePayerAccountCredential = new ArrayList<>();
+            caver = Caver.build(Caver.DEFAULT_URL);
+        }
+
+        public List<KlayCredentials> getTransactionAccountCredential() {
+            return transactionAccountCredential;
+        }
+
+        public List<KlayCredentials> getUpdateAccountCredential() {
+            return updateAccountCredential;
+        }
+
+        public List<KlayCredentials> getFeePayerAccountCredential() {
+            return feePayerAccountCredential;
+        }
+
+        public List<KlayCredentials> getSenderCredentialForTest(boolean isUpdateTest) {
+            if (isUpdateTest) {
+                return updateAccountCredential;
+            }
+            return transactionAccountCredential;
+        }
+
+        public KlayCredentials getOldAccount() {
+            return oldAccount;
+        }
+
+        public AccountKey getNewAccountKey() {
+            return newAccountKey;
+        }
+
+        public String getAddress() {
+            return address;
+        }
+
+        public void generateTestAccount(int transactionAccountCount, int updateAccountCount, int feePayerAccountCount) throws Exception {
+            oldAccount = createAccount();
+            address = oldAccount.getAddress();
+
+            setRandomRoleBasedNewAccountKey(oldAccount, transactionAccountCount, updateAccountCount, feePayerAccountCount);
+
+            TransactionManager transactionManager = new TransactionManager.Builder(caver, oldAccount)
+                    .setTransactionReceiptProcessor(new PollingTransactionReceiptProcessor(caver, 1000, 10))
+                    .setChaindId(LOCAL_CHAIN_ID)
+                    .build();
+
+            AccountUpdateTransaction accountUpdateTx = AccountUpdateTransaction.create(
+                    oldAccount.getAddress(),
+                    newAccountKey,
+                    GAS_LIMIT
+            );
+
+            KlayTransactionReceipt.TransactionReceipt transactionReceipt = transactionManager.executeTransaction(accountUpdateTx);
+            assertEquals("0x1", transactionReceipt.getStatus());
+
+            KlayAccountKey klayAccountKey = caver.klay().getAccountKey(oldAccount.getAddress(), DefaultBlockParameterName.LATEST).send();
+            AccountKey responseAccountKey = klayAccountKey.getResult().getKey();
+
+            assertEquals("Response\n" + responseAccountKey.toString() + "\nExpected" + newAccountKey.toString(), responseAccountKey, newAccountKey);
+        }
+
+        public void setRandomRoleBasedNewAccountKey(KlayCredentials oldAccount, int transactionAccountCount, int updateAcocountCount, int feePayerAccountCount) throws Exception {
+            List<AccountKey> roleBasedAccountKeyList = new ArrayList<>();
```

Comment:
You made bunch of methods to support extended `Klaycredentials`. I think this test is good opportunity to test the bunch of methods whether those are working well.